From 8f5f058ce80654270d6bad41562d8da5cf1d34eb Mon Sep 17 00:00:00 2001
From: Gerrit Pape <pape@smarden.org>
Date: Thu, 3 May 2007 09:40:38 +0000
Subject: [PATCH] arm/syscalls.h: fix __ARGS_* according to comment.

Comment states
/* ok the next few values are for the optimization of the unified syscalls
 * on arm.
 * If the syscall has #arguments
 *      <=4 set to 0
 *       >4 set to 1
 *
 * Since the majority of the syscalls need <=4 arguments this saves a lot
 * of byte (12 per syscall) and cycles (~16)
 */
so adjust __ARGS_* to be either 0 or 1.
---
 arm/syscalls.h |   16 ++++++++--------
 1 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/arm/syscalls.h b/arm/syscalls.h
index a5fa27a..22fd782 100644
--- a/arm/syscalls.h
+++ b/arm/syscalls.h
@@ -597,15 +597,15 @@
 #define __ARGS_mq_getsetattr		0
 #define __ARGS_waitid			0
 
-#define __ARGS_add_key			5
-#define __ARGS_request_key		4
-#define __ARGS_keyctl			3
-#define __ARGS_vserver			3
-#define __ARGS_ioprio_set		3
-#define __ARGS_ioprio_get		2
+#define __ARGS_add_key			1
+#define __ARGS_request_key		0
+#define __ARGS_keyctl			0
+#define __ARGS_vserver			0
+#define __ARGS_ioprio_set		0
+#define __ARGS_ioprio_get		0
 #define __ARGS_inotify_init		0
-#define __ARGS_inotify_add_watch	3
-#define __ARGS_inotify_rm_watch		2
+#define __ARGS_inotify_add_watch	0
+#define __ARGS_inotify_rm_watch		0
 
 
 #ifdef __ASSEMBLER__
-- 
1.5.1.3

